BinaryTree.java

```diff
@@ -0,0 +1,137 @@
+import java.util.InputMismatchException;
+
+/**
+ * Created by basescu on 29.05.17.
+ */
+public class BinaryTree {
+    private Integer info;
+    private BinaryTree left;
+    private BinaryTree right;
+
+    public BinaryTree(int info, BinaryTree left, BinaryTree right) {
+        this.info = info;
+        this.left = left;
+        this.right = right;
+    }
+
+    public Integer getInfo() {
+        return info;
+    }
+
+    public String printAncestors(Integer key) {
+        return this.printAncestors(key, "");
+    }
+
+    private String printAncestors(Integer key, String ancestors) {
+        if(this.info == key)
+            return ancestors;
+        if (this.left == null && this.right == null)
+            return "";
+        StringBuilder sb = new StringBuilder(ancestors.length() + 10);
+        sb.append(this.info);
+        sb.append(" ");
+        sb.append(ancestors);
+        String left = "";
+        String right = "";
+
+        if(this.left != null) {
+           left = this.left.printAncestors(key, sb.toString());
+        }
+        if(this.right != null) {
+            right = this.right.printAncestors(key, sb.toString());
+        }
+
+        if(left != "")
+            return left;
+        return right;
+    }
+
+    private BinaryTree lowestCommonAncestor(BinaryTree tree1, BinaryTree tree2) {
```

Comment:
Using BinaryTree as inputs to this method is quite inconvenient and potentially confusing - a person might think that they can pass multi-level trees here and that algorithm will match them against the tree contained in this. For example if this contains a tree (1, (2, null, null), (3, (4, null, null), (5, null, null))) one might try to find common ancestor of (3, (2, null, null), null) (note, that such subtree does not exist in our main tree) and (2, null, null), and would be surprised that common ancestor is found even though first subtree does not exist.

Since only info of the root node of each parameter tree is used, why not just use int info1 and int info2 here?